pkg/sypgp/sypgp.go

```diff
@@ -741,136 +795,166 @@ func ExportPubKey(kpath string, armor bool) error {
 }
 
 // ImportPrivateKey Will import a private key from a file (kpath).
-func ImportPrivateKey(kpath string) error {
+func ImportPrivateKey(entity *openpgp.Entity) error {
 	// Load the local private keys as entitylist
+
 	privateEntityList, err := LoadPrivKeyring()
 	if err != nil {
 		return err
 	}
-	// Load the private key as an entitylist
-	pathEntityList, err := LoadKeyringFromFile(kpath)
-	if err != nil {
-		return fmt.Errorf("unable to get entity from: %s: %v", kpath, err)
-	}
+
 	// Get local keyring (where the key will be stored)
 	secretFilePath, err := os.OpenFile(SecretPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
 	if err != nil {
 		return err
 	}
 	defer secretFilePath.Close()
 
-	// Go through the keystore checking for the given fingerprint
-	for _, pathEntity := range pathEntityList {
-		isInStore := false
-
-		for _, privateEntity := range privateEntityList {
-			if privateEntity.PrimaryKey.Fingerprint == pathEntity.PrimaryKey.Fingerprint {
-				isInStore = true
-				break
-			}
+	isInStore := false
 
+	for _, privateEntity := range privateEntityList {
+		if privateEntity.PrimaryKey.Fingerprint == entity.PrimaryKey.Fingerprint {
+			isInStore = true
+			break
 		}
 
-		if !isInStore {
-			// Make a clone of the entity
-			newEntity := &openpgp.Entity{
-				PrimaryKey:  pathEntity.PrimaryKey,
-				PrivateKey:  pathEntity.PrivateKey,
-				Identities:  pathEntity.Identities,
-				Revocations: pathEntity.Revocations,
-				Subkeys:     pathEntity.Subkeys,
-			}
+	}
 
-			// Check if the key is encrypted, if it is, decrypt it
-			if pathEntity.PrivateKey.Encrypted {
+	if !isInStore {
+		// Make a clone of the entity
+		newEntity := &openpgp.Entity{
+			PrimaryKey:  entity.PrimaryKey,
+			PrivateKey:  entity.PrivateKey,
+			Identities:  entity.Identities,
+			Revocations: entity.Revocations,
+			Subkeys:     entity.Subkeys,
+		}
+
+		// Check if the key is encrypted, if it is, decrypt it
+		if entity.PrivateKey != nil {
+			if entity.PrivateKey.Encrypted {
 				err = DecryptKey(newEntity, "Enter your old password : ")
 				if err != nil {
 					return err
 				}
 			}
+		} else {
+			sylog.Fatalf("Could not import the private key: wrong format")
```

Comment:
This should return a error instead, like:

```suggestion
			return fmt.Errorf("corrupted key, unable to recover data")
```